### PR TITLE
 Centaur reader improvements

### PR DESCRIPTION
**Pull request by Laar**
Improvements in the Preprocessor as first part of #61 

 - Read zone information from centaur meshes
 - Documentation on the centaur mesh format
 - Introduce `MeshSource` to clarify the interface the mesh reader expects from the mesh

TODO:

 - [x] Add zone info to `MeshSource`
 - [x] Add zone info to the preprocessor mesh

### CODE REVIEW

File: applications/Preprocessor/MeshSource.h [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+#define HPGEM_MESHSOURCE_H
+
+#include <vector>
+#include "customIterator.h"
+
+namespace Preprocessor {
+
+/**
+ * Source of a mesh
+ */
```
Comment (JensWehner) [inline applications/Preprocessor/MeshSource.h:48]: This explanation could be a bit more verbose. :)

Comment (Laar) [reply applications/Preprocessor/MeshSource.h]: It should be a bit better now

Comment (JensWehner) [reply applications/Preprocessor/MeshSource.h]: It is much better

**Review by JensWehner — APPROVED** [review]
LGTM

**Pull request merged**